nibabel/externals/netcdf.py

```diff
@@ -599,18 +606,20 @@ def _read_var_array(self):
             else:  # not a record variable
                 # Calculate size to avoid problems with vsize (above)
                 a_size = reduce(mul, shape, 1) * size
-                if self.use_mmap:
+                if self.file_bytes >= 0 and begin_+a_size > self.file_bytes:
+                    data = fromstring(b'\x00'*a_size, dtype=dtype_)
+                elif self.use_mmap:
                     mm = mmap(self.fp.fileno(), begin_+a_size, access=ACCESS_READ)
                     data = ndarray.__new__(ndarray, shape, dtype=dtype_,
                             buffer=mm, offset=begin_, order=0)
                 else:
                     pos = self.fp.tell()
                     self.fp.seek(begin_)
-                    data = fromstring(self.fp.read(a_size), dtype=dtype_)
-                    # Avoid exception for empty variables.
-                    if data.shape != (0,):
-                        data.shape = shape
-
+                    buf = self.fp.read(a_size)
```

Comment:
Comment here?  I guess this is for a broken file where there's an expected variable truncated at the end?  Do you really want to allow that case without raising an error?